ComfortZoneApp/app/src/main/java/com/example/comfortzone/profile/ui/ProfileFragment.java

```diff
@@ -82,8 +87,27 @@ private void setUpRecyclerView() {
     }
 
     private void getSavedCities() {
-        ArrayList<Integer> savedCityIds = (ArrayList<Integer>) currentUser.get(KEY_SAVED_CITIES);
-        AllWeathersDatabase db = AllWeathersDatabase.getDbInstance(getContext());
-        adapter.addAll(savedCityIds.stream().map(cityId -> db.weatherDao().getWeatherById(cityId)).collect(Collectors.toList()));
+        Observable<Object> dataSetupObservable = WeatherDbUtil.maybeUpdateCitiesList(getActivity());
+        Subscriber dataSetupSubscriber = new Subscriber() {
+            @Override
+            public void onCompleted() {
+                requireActivity().runOnUiThread(new Runnable() {
```

Comment:
As mentioned, we should always check nullability of `requireActivity` especially in callback, because we don't know when fragment will lose its hosting activity